File: packages/shared-types/src/alerts.ts
export type AlertSeverity = "clear" | "inform" | "watch" | "warning" | "act_now";

export const SEVERITY_ORDER: AlertSeverity[] = [
  "clear",
  "inform",
  "watch",
  "warning",
  "act_now",
];

export function severityIndex(s: AlertSeverity): number {
  return SEVERITY_ORDER.indexOf(s);
}

export function isAtLeast(a: AlertSeverity, b: AlertSeverity): boolean {
  return severityIndex(a) >= severityIndex(b);
}

export interface AlertSource {
  id: string;
  name: string;
  url?: string;
  timestamp: string;
  official: boolean;
}

export interface NormalizedAlert {
  id: string;
  raw_alert_id?: string;
  headline: string;
  location_label: string;
  hazard_type: string;
  severity: AlertSeverity;
  starts_at?: string;
  ends_at?: string;
  summary: string;
  description: string;
  instruction?: string;
  recommended_actions: string[];
  source: AlertSource;
  is_active: boolean;
  fetched_at: string;
  geometry?: import("geojson").Geometry | null;
}

export interface AlertSummary
  extends Pick<
    NormalizedAlert,
    | "id"
    | "headline"
    | "location_label"
    | "hazard_type"
    | "severity"
    | "starts_at"
    | "ends_at"
    | "summary"
    | "is_active"
  > {}

export interface AlertsResponse {
  count: number;
  alerts: NormalizedAlert[];
}

File: apps/api/app/schemas/alerts.py
from datetime import datetime
from enum import Enum
from typing import Any, Dict, List, Optional
from uuid import UUID

from pydantic import BaseModel, Field


class AlertSeverity(str, Enum):
    CLEAR = "clear"
    INFORM = "inform"
    WATCH = "watch"
    WARNING = "warning"
    ACT_NOW = "act_now"


class AlertSource(BaseModel):
    id: str
    name: str
    url: Optional[str] = None
    timestamp: datetime
    official: bool = True


class NormalizedAlert(BaseModel):
    id: UUID
    raw_alert_id: Optional[str] = None
    headline: str
    location_label: str
    hazard_type: str
    severity: AlertSeverity
    starts_at: Optional[datetime] = None
    ends_at: Optional[datetime] = None
    summary: str
    description: str
    instruction: Optional[str] = None
    recommended_actions: List[str] = Field(default_factory=list)
    source: AlertSource
    is_active: bool = True
    fetched_at: datetime
    geometry: Optional[Dict[str, Any]] = None


class RawAlert(BaseModel):
    id: UUID
    source_name: str
    external_id: str
    payload_json: Dict[str, Any]
    fetched_at: datetime
    processed: bool = False


class AlertSummaryResponse(BaseModel):
    """Lightweight projection for list views."""
    id: UUID
    headline: str
    location_label: str
    hazard_type: str
    severity: AlertSeverity
    starts_at: Optional[datetime] = None
    ends_at: Optional[datetime] = None
    summary: str
    is_active: bool


class AlertDetailResponse(NormalizedAlert):
    pass
